packages/hint/src/lib/utils/dom/get-element-by-url.ts

```diff
@@ -0,0 +1,51 @@
+import { URL } from 'url';
+
+import { HTMLDocument, HTMLElement } from '../../types/html';
+
+const getSrcsetUrls = (srcset: string): string[] => {
+    if (!srcset) {
+        return [];
+    }
+
+    const parts = srcset.split(',');
+    const urls = parts.reduce((total, part) => {
+        const url = part.trim().split(' ')[0];
+
+        if (!url) {
+            return total;
+        }
+
+        total.push(url.trim());
+
+        return total;
+    }, [] as string[]);
+
+    return urls;
+};
+
+export default (dom: HTMLDocument, url: string): HTMLElement | null => {
+    const elements = dom.querySelectorAll('[href],[src],[poster],[srcset]').filter((element: any) => {
+        const elementUrl = element.getAttribute('href') || element.getAttribute('src') || element.getAttribute('poster');
```

Comment:
I don't think this is going to be called more than once with the same url (if so, is because the user has something duplicated is the code). But we can cache the result of the querySelectorAll.

Reply:
Probably a premature optimization at this point. I'd wait.

Reply:
We might end up creating the same `new Url` quite a few times depending on how many links the page has.
Can I have a `//TODO` here and hopefully one day we will do #210 ?

Reply:
> We might end up creating the same new Url quite a few times depending on how many links the page has.

We shouldn't have the same url repeated many often, the only scenario I can think about is when we have the same image in few places.

I will add the //TODO